src/common/cuepoint/cuepoint-manager.js

```diff
@@ -44,13 +44,16 @@ export class CuePointManager {
   }
 
   getCuePointById(id: string): ?TextTrackCue {
+    let cuePointById = null;
     const metadataTracks = this._getMetadataTracks();
-    metadataTracks.forEach(track => {
+    metadataTracks.every(track => {
       const cuePoint = track.cues.getCueById(id);
       if (cuePoint) {
-        return cuePoint;
+        cuePointById = cuePoint;
       }
+      return !cuePoint;
     });
+    return cuePointById;
```

Comment:
```getCuePointById(id: string): ?TextTrackCue {
    const metadataTracks = this._getMetadataTracks();
    return metadataTracks.find(track => {
      return track.cues.getCueById(id);
    })
```

Reply:
We think this is clearer :)

Reply:
`metadataTracks.find` will return the track not the cue

Reply:
you are right